pilot/pkg/model/push_context.go

```diff
@@ -2095,7 +2088,7 @@ func (ps *PushContext) initGateways(env *Environment) error {
 }
 
 func (ps *PushContext) initAmbient(env *Environment) {
-	ps.AmbientSnapshot = env.AmbientSnapshot()
+	ps.ambientIndex = env
```

Comment:
It is suspicious to me, previously(i mean before ambient merged upstream) push context only maintains the indexes of configs/services for a push epic, but now we are using env, calling get/list may get the configs/services of latest which is different with the time xds push triggered.

I am not sure whether it will influence xds cache

Reply:
Its the same as EndpointShards basically